app/models/manageiq/providers/redhat/infra_manager/vm/reconfigure.rb

```diff
@@ -39,4 +40,103 @@ def spec_for_added_disks(disks)
+      :add    => (network_adapters_add(options[:network_adapter_add]) if options[:network_adapter_add]),
+      :remove => (network_adapters_remove(options[:network_adapter_remove]) if options[:network_adapter_remove])
+    }.compact
+  end
+
+  def network_adapters_add(adapters)
+    nic_names = hardware.nics.pluck(:device_name)
+    switch_ids = HostSwitch.where(:host_id => host.id).pluck(:switch_id)
+
+    adapters.collect do |adapt|
+      new_nic_name = suggest_nic_name(nic_names)
+      nic_names << new_nic_name
+      network_adapter_add_spec(adapt['network'],
+                               new_nic_name,
+                               switch_ids)
+    end
+  end
+
+  def network_adapters_edit(adapters)
+    switch_ids = HostSwitch.where(:host_id => host.id).pluck(:switch_id)
+
+    adapters.collect do |adapt|
+      network_adapter_edit_spec(adapt['network'],
+                                adapt['name'],
+                                switch_ids)
+    end
+  end
+
+  def network_adapters_remove(adapters)
+    adapters.collect do |adapt|
+      network_adapter_remove_spec(adapt['network']['vlan'],
+                                  adapt['network']['name'],
+                                  adapt['network']['mac'])
+    end
+  end
+
+  def network_adapter_add_spec(network_name, nic_name, switch_ids)
+    lan = find_lan_by_name(network_name, switch_ids)
+    raise MiqException::MiqVmError, "Network [#{network_name}] not available for the target" unless lan
+
+    {
+      :network         => network_name,
+      :name            => nic_name,
+      :vnic_profile_id => lan.uid_ems
+    }
+  end
+
+  def network_adapter_edit_spec(network_name, nic_name, switch_ids)
+    nic = hardware.nics.find_by_name(nic_name)
+    raise MiqException::MiqVmError, "No NIC named '#{nic_name}' was found" unless nic
+
+    lan = find_lan_by_name(network_name, switch_ids)
+    raise MiqException::MiqVmError, "Network [#{network_name}] not available for the target" unless lan
+
+    {
+      :network         => network_name,
+      :name            => nic_name,
+      :vnic_profile_id => lan.uid_ems,
+      :nic_id          => nic.uid_ems
+    }
+  end
+
+  def network_adapter_remove_spec(network_name, nic_name, mac)
+    nic = hardware.nics.find_by_name(nic_name)
+    raise MiqException::MiqVmError, "No NIC named '#{nic_name}' was found" unless nic
+
+    {
+      :network     => network_name,
+      :name        => nic_name,
+      :mac_address => mac,
+      :nic_id      => nic.uid_ems
+    }
+  end
+
+  def find_lan_by_name(network_name, switch_ids)
+    lan_name, ext_switch_name = network_name.split('/').collect(&:strip)
+
+    switch_ids = ext_switch_ids_by_name(ext_switch_name) if ext_switch_name
+
+    Lan.find_by(:name => lan_name, :switch_id => switch_ids)
+  end
+
+  def ext_switch_ids_by_name(switch_name)
+    ext_management_system.external_distributed_virtual_switches.where(:name => switch_name).pluck(:id)
```

Comment:
@gekorob is the name of the DVS unique across the whole EMS?

Reply:
Aka is it possible to have two Lans with the same name on two switches with the same name

Reply:
@agrare in RHV is not possible to add a new logical network with the same name of another logical network in the same datacenter

Reply:
:+1:

Reply:
@agrare I made some experiments and your first doubt was correct. I mean what I wrote about RHV is correct, but within the same EMS we can have more datacenters.
The problem is that we have external_distributed_virtual_switches connected to the ems and that, I think, is not correctly modelling the RHV situation

Reply:
@borod108 is there a way to get external_distributed_virtual_lans for the datacenter of the vm under reconfiguration (without asking to the provider)? or am I missing something?

Reply:
Okay so hosts under a datacenter are not connected to switches from another datacenter?

That was why I suggested just creating host_switches join table records for every host to the distributed switches so that the rest of the modeling would "Just work" but I think there were some issues with that that maybe @borod108 remembers.  Something to do with targeted refresh but I think this could be overcome.

Reply:
Will it be possible to have a join table between datacenter and switches? In particular to have external_distributed_virtual_switches connected to datacenter?

Reply:
Okay so now that we have a relationship between external switches and their owning datacenter we can use that here right?

Reply:
Yes I'm about to push the commits with tests

Reply:
I was testing also on a RHV env

Reply:
:heart: awesome